pkg/config/config.go

```diff
@@ -32,6 +32,22 @@ type ComponentSettings struct {
 	// The builder image to use
 	ComponentType *string `yaml:"ComponentType,omitempty"`
 
+	// Path is path to binary in current/context dir
+	Path *string `yaml:"Path,omitempty"`
+
+	// Ref is component source git ref but can be levaraged for more in future
+	Ref *string `yaml:"Ref,omitempty"`
+
+	// Type is type of component source: git/local/binary
+	SourceType *string `yaml:"SourceType,omitempty"`
```

Comment:
also, @kadel was of the opinion that this could be of type https://github.com/redhat-developer/odo/blob/master/pkg/occlient/occlient.go#L67

Reply:
aah i think i see the magic below

Reply:
@anmolbabu there is a `CreateType` in utils package as well

Reply:
done resolved, changed sourceType to create type